conda_forge_tick/migrators/nvtools.py

```diff
@@ -0,0 +1,244 @@
+
+    def filter(self, attrs: "AttrsTypedDict", not_bad_str_start: str = "") -> bool:
+        """If true don't act upon node
+
+        Parameters
+        ----------
+        attrs : dict
+            The node attributes
+        not_bad_str_start : str, optional
+            If the 'bad' notice starts with the string then it is not
+            to be excluded. For example, rebuild migrations don't need
+            to worry about if the upstream can be fetched. Defaults to ``''``
+
+        Returns
+        -------
+        bool :
+            True if node is to be skipped
+        """
+        return (
+            attrs["archived"]
+            or "https://developer.download.nvidia.com" not in attrs["source"]["url"]
+        )
+
+    def migrate(
+        self, recipe_dir: str, attrs: AttrsTypedDict, **kwargs: Any
+    ) -> MigrationUidTypedDict:
+        """Perform the migration, updating the ``meta.yaml``
+
+        Parameters
+        ----------
+        recipe_dir : str
+            The directory of the recipe
+        attrs : dict
+            The node attributes
+
+        Returns
+        -------
+        namedtuple or bool:
+            If namedtuple continue with PR, if False scrap local folder
+        """
+        # STEP 0: Bump the build number
+        self.set_build_number(os.path.join(recipe_dir, "meta.yaml"))
+
+        # STEP 1: Add cf-nvidia-tools to build requirements
+        meta = os.path.join(recipe_dir, "meta.yaml")
+        if _file_contains(meta, "cf-nvidia-tools"):
+            logging.debug("cf-nvidia-tools already in meta.yaml; not adding again.")
+        else:
+            _insert_requirements_build(meta)
+            logging.debug("cf-nvidia-tools added to meta.yaml.")
+
+        # STEP 2: Add check-glibc to the build script
+        build = os.path.join(recipe_dir, "build.sh")
+        if os.path.isfile(build):
+            if _file_contains(build, "check-glibc"):
+                logging.debug(
+                    "build.sh already contains check-glibc; not adding again."
+                )
+            else:
+                with open(build, "a") as file:
+                    file.write('\ncheck-glibc "$PREFIX"/lib/*.so.* "$PREFIX"/bin/*\n')
+                logging.debug("Added check-glibc to build.sh")
+        else:
+            if _file_contains(meta, "check-glibc"):
+                logging.debug(
+                    "meta.yaml already contains check-glibc; not adding again."
+                )
+            else:
+                _insert_build_script(meta)
+                logging.debug("Added check-glibc to meta.yaml")
+
+        # STEP 3: Remove os_version keys from conda-forge.yml
+        config = os.path.join(recipe_dir, "..", "conda-forge.yml")
+        with open(config) as f:
+            y = yaml_safe_load(f)
+        y_orig = copy.deepcopy(y)
+        y.pop("os_version", None)
+        if y_orig != y:
+            with open(config, "w") as f:
+                yaml_safe_dump(y, f)
```

Comment:
So we are dropping the `os_version` specification of the Docker image to use?

Wonder if it makes to add/confirm `{{ stdlib("c") }}` is used and `c_stdlib_version` is set correctly

Reply:
Yes. It's not necessary to force use an old container.

The tool fails if c_stdlib_version is not set, so indirectly this is already checked?